src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs

```diff
@@ -100,7 +118,7 @@ public TCommandResult Execute(IConnection connection, CancellationToken cancella
                 _session.WasUsed();
             }
 
-            switch (_responseHandling())
+            switch (message.ResponseHandling)
```

Comment:
Similar to OP_MSG, the value of RespsonseHandling *might* change from Ignore to Result if the write concern is changed from { w : 0 } to { w : 1 } by the PostWriteAction.